### PR TITLE
[videa] Add new extractor

### PR DESCRIPTION
**Pull request by gyaraki**
Extractor support for videa.hu, hungarian videosharing site.

### CODE REVIEW

File: youtube_dl/extractor/videa.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+class VideaIE(InfoExtractor):
+    _VALID_URL = r'https?://videa\.hu/videok/(.+?)/?(?P<id>\d+)'
+    IE_NAME = 'videa.hu'
+
+    _TEST = {
+        'url': 'http://videa.hu/videok/kreativ/igy-lehet-nekimenni-a-hosegnek-ballon-hoseg-CZqIVSVYfJKf9bHC',
+        'md5': 'e9a24350bbd485746d23519319949e72',
+        'info_dict': {
+            'id': 'CZqIVSVYfJKf9bHC',
+            'ext': '1150387',
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: Invalid extension.

File: youtube_dl/extractor/videa.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            'title': 'Így lehet nekimenni a hőségnek',
+            'description': '',
+            'uploader': 'viktoratakarító',
+        } 
+    }
+
+    def _real_extract(self, url):
+
+        # the last part of the slug is the real video hash id
+        video_id = self._search_regex(r'([A-Za-z0-9]+)$', url, 'video id')
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: This should be extracted from `_VALID_URL` with `self._match_id`.

File: youtube_dl/extractor/videa.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        # access the oembed data for video infos
+        info = self._download_json(
+            'http://videa.hu/oembed/?url=%s&format=json' % info_data, url, 'Downloading video info')
+
+        # get the player xmldata for the video file url 
+        flvplayer = self._download_xml(
+            'http://videa.hu/flvplayer_get_video_xml.php?v=%s' %  video_id, 'Downloading player datas')
+        
+        duration = flvplayer.find('./video/duration').text;
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: Unused.

File: youtube_dl/extractor/videa.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        values = flvplayer.find('./video/versions').text;
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: Unused.

File: youtube_dl/extractor/videa.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        for elem in flvplayer.findall('video/versions/version'):
+            video_url = elem.get('video_url')
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: All video formats must be extracted.

File: youtube_dl/extractor/videa.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+        return {
+            'id': video_id,
+            'title': info['title'],
+            'url': video_url,
+            'thumbnail': info['thumbnail_url'],
+            'description': '',
+            'uploader': info['author_name'],
+        }
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: No newline at the end of file.

File: youtube_dl/extractor/videa.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        duration = flvplayer.find('./video/duration').text;
+        values = flvplayer.find('./video/versions').text;
+        for elem in flvplayer.findall('video/versions/version'):
+            video_url = elem.get('video_url')
+
+        return {
+            'id': video_id,
+            'title': info['title'],
+            'url': video_url,
+            'thumbnail': info['thumbnail_url'],
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: Extraction must be tolerate to missing fields everything apart from title and formats is optional.

File: youtube_dl/extractor/videa.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        video_id = self._search_regex(r'([A-Za-z0-9]+)$', url, 'video id')
+
+        info_data = compat_urllib_parse.urlencode({
+            'format': 'json',
+            'url': url,
+        })
+
+        # access the oembed data for video infos
+        info = self._download_json(
+            'http://videa.hu/oembed/?url=%s&format=json' % info_data, url, 'Downloading video info')
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: This results in `http://videa.hu/oembed/?url=url=http%3A%2F%2Fvidea.hu%2Fvideok%2Fkreativ%2Figy-lehet-nekimenni-a-hosegnek-ballon-hoseg-CZqIVSVYfJKf9bHC&format=json&format=json` that is obviously not what was intended. Second argument must be `video_id`.

File: youtube_dl/extractor/videa.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+        # the last part of the slug is the real video hash id
+        video_id = self._search_regex(r'([A-Za-z0-9]+)$', url, 'video id')
+
+        info_data = compat_urllib_parse.urlencode({
+            'format': 'json',
+            'url': url,
+        })
+
+        # access the oembed data for video infos
```
Comment (dstftw) [inline youtube_dl/extractor/videa.py]: Avoid senseless comments.

**Comment by yan12125** [comment]
Since #8497 landed, please move changes in ```youtube_dl/extractors/__init__.py``` to ```youtube_dl/extractors/extractors.py```. Check step 5 of the new [developer instructions](https://github.com/rg3/youtube-dl/blob/master/README.md#adding-support-for-a-new-site) for more information.

**Pull request closed without merging**